### ISSUE TITLE
API Proposal: Broadcasts with Layout

### ISSUE THREAD

**Issue by aoberoi** [opentok/opentok-node#128]
This is a request for comments regarding the API for upcoming platform functionality. The platform functionality is not generally available at the moment and there is no guarantee it will be released in any form.
## Broadcasts

All `callback` functions have the signature `fn(err, result)`
### Create a Broadcast

``` javascript
// each callback has a `Broadcast` instance as the result

// simplest case
opentok.startBroadcast(sessionId, callback);

// using a predefined layout
opentok.startBroadcast(sessionId, {
  layout: 'pip',
}, callback);

// using custom layout
let broadcast = opentok.startBroadcast(sessionId, {
  layout: {
    stylesheet: '{css string}'
  }
}, callback);
```
### Stop a Broadcast

``` javascript
// each callback has a `Broadcast` instance as the result

// without an instance of `Broadcast`
opentok.stopBroadcast(broadcastId, callback);

broadcast.stop(callback);
```
### Get a Broadcast

``` javascript
// callback has a `Broadcast` instance as the result

opentok.getBroadcast(broadcastId, callback);

// properties of a `Broadcast` instance
broadcast.id; // string
broadcast.sessionId; // string
broadcast.createdAt; // number of seconds since unix epoch
broadcast.updatedAt; // number of seconds since unix epoch
broadcast.partnerId; // string
broadcast.urls // object
broadcast.urls.hls // string
```
### Get a Broadcast Layout

``` javascript
// each callback has an object with the shape { type: '', stylesheet: '' } as the result

// without an instance of `Broadcast`
opentok.getBroadcastLayout(broadcastId, callback);

broadcast.getLayout(callback);
```
### Update a Broadcast Layout

``` javascript
// each callback has an object with the shape { type: '', stylesheet: '' } as the result
// the `layout` parameter is similar to the value in the `'layout'` key of the options in `opentok.startBroadcast()`

// without an instance of `Broadcast`
opentok.updateBroadcastLayout(broadcastId, callback);

broadcast.updateLayout(layout)
```
### Update a Stream's Layout Class List

```
// each callback has no result
// TODO: should there be a result which describes the stream?

session.updateStream(streamId, {
  layoutClassList: 'class1'
}, callback);

opentok.updateStream(streamId, {
  layoutClassList: ['class1', 'class2']
}, callback);
```

**Comment by aoberoi** [comment]
we plan to begin an implementation next week. please comment at your earliest convenience.

this is related to #126 

Question: should we have an API for updating multiple streams?

**Comment by msach22** [comment]
This has been released with `v2.8.0`
